services/topology-engine/queue-engine/topologylistener/message_utils.py

```diff
@@ -189,15 +189,18 @@ def build_one_switch_flow_from_db(switch, stored_flow, output_action):
     return flow
 
 
-def build_delete_flow(switch, flow_id, cookie, meter_id, in_port, in_vlan, out_port):
+def build_delete_flow(node):
     flow = Flow()
     flow.clazz = "org.openkilda.messaging.command.flow.RemoveFlow"
     flow.transaction_id = 0
-    flow.flowid = flow_id
-    flow.cookie = cookie
-    flow.switch_id = switch
-    flow.meter_id = meter_id
-    flow.criteria = {'cookie': cookie, 'in_port': in_port, 'in_vlan': in_vlan, 'out_port': out_port}
+    flow.flowid = str(node['flow_id'])
+    flow.cookie = node['cookie']
+    flow.switch_id = str(node['switch_id'])
+    flow.meter_id = node['meter_id']
+
+    flow.criteria = {'cookie': node['cookie'], 'in_port': node['in_port'], 'in_vlan': node['in_vlan']}
+    if node['in_port'] != node['out_port']:
```

Comment:
To avoid confusion from this check we should add switch_id equality check there.

Reply:
I'll add a comment instead as discussed